server/workflow/workflow_server.go

```diff
@@ -242,39 +242,27 @@ func (s *workflowServer) ResubmitWorkflow(ctx context.Context, req *workflowpkg.
 
 func (s *workflowServer) ResumeWorkflow(ctx context.Context, req *workflowpkg.WorkflowResumeRequest) (*v1alpha1.Workflow, error) {
 	wfClient := auth.GetWfClient(ctx)
-	_, err := s.getWorkflowAndValidate(ctx, req.Namespace, req.Name, metav1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-
-	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), s.offloadNodeStatusRepo, req.Name, req.NodeFieldSelector)
+	wf, err := s.getWorkflowAndValidate(ctx, req.Namespace, req.Name, metav1.GetOptions{})
 	if err != nil {
-		log.Warnf("Failed to resume %s: %+v", req.Name, err)
 		return nil, err
 	}
 
-	wf, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Get(req.Name, metav1.GetOptions{})
+	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().WorkflowOps(req.Namespace), wf, req.NodeFieldSelector)
```

Comment:
challenge one: now decoupled - we cannot determine if the op was valid

Reply:
This is a potential issue. At the moment my client code retries the 'resume' a few times on failure, as it's possible the resume node hasn't been created yet.